soh/soh/GameMenuBar.cpp

```diff
@@ -811,6 +813,12 @@ namespace GameMenuBar {
                         " - Boss keys: Vanilla size and texture\n"
                         " - Skulltula Tokens: Small skulltula chest\n"
                     );
+                    if (CVar_GetS32("gChestSizeAndTextureMatchesContents", 0) > 0) {
+                        UIWidgets::PaddedEnhancementCheckbox("Chests of Agony", "gChestSizeDependsStoneOfAgony", true, false);
+                        UIWidgets::Tooltip("Only change the size/texture of chests if you have the Stone of Agony.");
+                    } else {
+                        CVar_SetS32("gChestSizeDependsStoneOfAgony", 0);
```

Comment:
Currently this line will run every frame, instead we need to do an `if` around the ` UIWidgets::EnhancementCombobox("gChestSizeAndTextureMatchesContents"` above, which will only update this when that value is changed

Reply:
All interactable ImGui elements should return a bool that returns true on the frame that the input was interacted with. I've updated some of our UIWidgets to expose this as it should be, but it looks like `EnhancementCombobox` and a few others have not exposed this yet.

Anywhere else this pattern appears (I see it is indeed like this for Kokiri Draw Distance) is a not desired, and should also be updated to only be set on user interaction.

Reply:
I can push up a PR fixing the kokiri draw distance instance and exposing the bool via the checkbox, or you can add the exposing the bool in this PR, and I can submit a follow up fixing the kokiri draw distance

Reply:
> I can push up a PR fixing the kokiri draw distance instance and exposing the bool via the combobox, or you can add the exposing the bool in this PR, and I can submit a follow up fixing the kokiri draw distance

My bad for this one, didn't knew doing this would be an every frame check

Reply:
I went ahead and fixed Kokiri Draw as well as exposing the bool via combobox in 3e44805d.

Let it be known, I'm not a C++ dev; I work mostly with Python. So, if there are any other conventions or things I'm missing, let me know and I'll be happy to improve :)